src/loaders/ObjectLoader.js

```diff
@@ -610,6 +617,94 @@ THREE.ObjectLoader.prototype = {
 
 			}
 
+			/*
+			 * Now SkinnedMesh could have two bone sets in its children,
+			 * one is made from geometry.bones in SkinnedMesh constroctor
+			 * and another one is made from data.children in this method.
+			 * These two bone sets correspond to the same bones.
+			 * The former would represent a default stand pose while
+			 * the latter would represent a pose updated by skinning.
+			 * To let SkinnedMesh be updated pose, here this logic copies
+			 * the latter one to the former one then removes the latter one.
+			 * If in the latter one there're objects which don't correspond to
+			 * the ones in the former one, they'll be moved to the former one.
+			 */
```

Comment:
> what do you guys think about having 2 sets of bones?

There are two different bone representations used in games and editors.  The first is using nodes as bones.  This is used by editors who allow you to create animations - Clara.io, 3DS Max, Maya, etc.  This is because having them as nodes allows you to animate them, create bone hierarchies etc.  The other representation is to have them within a "character", this is how UE4, Unity3D has them.  Usually the in-character bone representations are optimized, they are light weight, and not editable because they are focused on playback.  ThreeJS is having both right now - it isn't wrong but presents a maintenance issue because now you have to maintain both and you have synchronization issues between the two.  I would suggest following UE4 and Unity3D in having only the in-character bone representation (e.g. skeleton) rather than the more costly (albeit more flexible) scene graph node-based approach.

The reason that people generally do not use the node-based approach in game engines is that usually the animations are all fixed to the names of the bones anyhow, thus no one generally modifies the bones at run-time.  Even for things like growing tenticals in games, all the bones are already there, they just start really small and extend those bones.  It is also nearly impossible to do real-time skin weights (which would need to be redone if you are changing your bones) that are sufficient quality.